src/rewriter.rs

```diff
@@ -171,18 +246,70 @@ fn parse_js(
     )
 }
 
-fn transform_js<R: Read>(
+fn apply_transformation_passes(
+    mut program: Program,
+    config: &mut Config,
+    passes: &mut HashSet<TransformPass>,
+    meta: FileMeta,
+) -> Result<(Program, TransformStatus)> {
+    let mut transform_status = TransformStatus::not_modified(config);
+
+    if let Some(pass_to_remove) = passes
+        .iter()
+        .find(|&&pass| {
+            pass(&mut program, &mut transform_status, config, &meta);
+            transform_status.status == Status::Cancelled
+        })
+        .cloned()
+    {
+        passes.remove(&pass_to_remove);
+        if !passes.is_empty() && !config.strict {
+            transform_status.status = Status::Restart
+        }
+    }
+    Ok((program, transform_status))
+}
+
+fn transform_iast(
+    program: &mut Program,
+    transform_status: &mut TransformStatus,
+    config: &mut Config,
+    _meta: &FileMeta,
+) {
+    let mut block_transform_visitor = TaintBlockTransformVisitor::default(transform_status, config);
+    program.visit_mut_with(&mut block_transform_visitor);
+}
+
+fn transform_orchestrion(
+    program: &mut Program,
+    transform_status: &mut TransformStatus,
+    config: &mut Config,
+    meta: &FileMeta,
+) {
+    if let Some(instrumentor) = &mut config.instrumentor {
```

Comment:
Can't we do this check before ? Because if for instance orchestrion is the only pass to run, if this check is False, it is useless to parse the file beforehand.

Reply:
We don't really know what state the config is going to be in prior to this.